app/Controllers/Timetable.hs

```diff
@@ -149,14 +153,16 @@ pullDatabase (code, section, session) = runDb $ do
 type SystemTime = String
 
 -- | Creates all the events for a course.
-getEvents :: SystemTime -> MeetTime' -> IO [String]
-getEvents systemTime lect = do
-    courseInfo <- getCourseInfo lect  -- Get the course information
-    let startTimes = third courseInfo   -- Extract start times
-        endTimes = fourth courseInfo     -- Extract end times
-        dates = fifth courseInfo         -- Extract dates
-    events <- mapM (eventsByDate courseInfo) (zip' startTimes endTimes dates)
-    return (concat events)
+getEvents :: SystemTime -> Maybe MeetTime' -> IO [String]
+getEvents _ Nothing = do
+    return []
+getEvents systemTime (Just x) = do
+            courseInfo <- getCourseInfo x  -- Get the course information
```

Comment:
This is good, but now fix the indentation here. Also pick a better variable name than `x`